spec/requests/memberships_controller_spec.rb

```diff
@@ -0,0 +1,118 @@
+
+    post "/groups/#{group.id}/memberships"
+    expect(Membership.where(group: group, user: user).count).to eql(1)
+
+    expect(response).to have_http_status(:redirect)
+  end
+
+  it 'archive active_member of group' do
+    membership = create(:membership, :newbie)
+    group = membership.group
+    login_as(group.leader.user)
+    Timecop.freeze do
+      put "/groups/#{group.id}/memberships/#{membership.id}/archive", xhr: true
+
+      expect(membership.reload.archived.today?).to be true
+    end
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'archive inactive_member of group' do
+    membership = create(:membership, :inactive)
+    login_as(membership.group.leader.user)
+
+    Timecop.freeze do
+      put "/groups/#{membership.group.id}/memberships/#{membership.id}/archive", xhr: true
+
+      expect(membership.reload.archived.today?).to be true
+    end
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'unarchive active_member of group with group leader' do
+    membership = create(:membership, :archived)
+    login_as(membership.group.leader.user)
+
+    put "/groups/#{membership.group.id}/memberships/#{membership.id}/unarchive", xhr: true
+
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'unarchive inactive_member of group' do
+    membership = create(:membership, :inactive, :archived)
+    login_as(membership.group.leader.user)
+
+    put "/groups/#{membership.group.id}/memberships/#{membership.id}/unarchive", xhr: true
+
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'forbidden archive of member' do
+    login_as(create(:user))
+    membership = create(:membership, :leader)
+    put "/groups/#{membership.group.id}/memberships/#{membership.id}/unarchive", xhr: true
+
+    expect(response).to have_http_status(:forbidden)
+  end
+
+  it 'inactivation of a group member' do
+    membership = create(:membership)
+    login_as(membership.group.leader.user)
+    Timecop.freeze do
+      post "/groups/#{membership.group.id}/memberships/#{membership.id}/inactivate", xhr: true
+
+      expect(membership.reload.end_date.today?).to be true
+    end
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'reactivation of an inactive member' do
+    membership = create(:membership, :inactive)
+    login_as(membership.group.leader.user)
+
+    post "/groups/#{membership.group.id}/memberships/#{membership.id}/reactivate", xhr: true
+
+    expect(membership.reload.end_date).to be_nil
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'delegation became false when inactivate a group member who delegated that group' do
+    user = create(:user, :who_delegated)
+    membership = user.primary_membership
+    login_as(membership.group.leader.user)
+
+    expect(membership.user.delegated).to be true
+    expect(membership.user.primary_membership).to eql(membership)
+
+    Timecop.freeze do
+      membership.id #do we need this?
```

Comment:
I found this line in the original test, and I kept it. It doesn't do anything, should I remove it?

Reply:
Yes please, it makes no sense.